ngi_pipeline/engines/qc_ngi/workflows.py

```diff
@@ -103,21 +104,39 @@ def workflow_fastqc(input_files, output_dir, config):
                 # Input file modified more recently than output file
                 fastq_to_analyze.add(fastq_file)
 
+
     num_threads = config.get("qc", {}).get("fastqc", {}).get("threads") or 1
     # Construct the command lines
     cl_list = []
-    if fastq_to_analyze:
-        safe_makedir(output_dir)
-        # Module loading
-        modules_to_load = get_all_modules_for_workflow("fastqc", config)
-        for module in modules_to_load:
-            cl_list.append("module load {}".format(module))
-        # Execute fastqc
+    # fastqc commands
+    for fastq_file in fastq_to_analyze:
+        #when building the fastqc command soflink in the qc_ngi folder the fastq file processed being sure to avoid name collision (i.e., same sample run in two different FC but on the same lane number). Run fastqc on the softlink and delete the soflink straight away. In case the fastq file does not match the regex force the exectuion of fastqc... this is what we have been always doing.
+        fc_id = re.search('\d{4,6}_(?P<fcid>[A-Z0-9]{10})', fastq_file).groups()[0]
+        m = re.match(r'([\w-]+).(fastq.*)', os.path.basename(fastq_file))
```

Comment:
This is a repetition of the same parsing/logic as above. I would prefer to put this in a function or variable in order not to repeat the logic.